python/tests/avro/test_resolver.py

```diff
@@ -164,17 +163,17 @@ def test_resolver_change_type() -> None:
 
 
 def test_promote_int_to_long() -> None:
-    assert promote(IntegerType(), LongType()) == IntegerReader()
+    assert promote(IntegerType(), LongType()) == LongType()
```

Comment:
Is it better to test `promote` or `resolve` here? I don't think it matters much.

Reply:
For those tests (and since it is in Avro), we want to use `resolve`. The difference is that with Resolve we check if the promotion is valid, but if we're going from float to double, we still want to read the float (since double is twice as many bytes), so we'll return a float.